drivers/adc/ad7490/README.rst

```diff
@@ -0,0 +1,214 @@
+started by using the **ad7490_start_seq** API, where either a shadow, control
+or consecutive type of sequencer cand be configured. (Please refer to the
+datasheet for each type of sequencer).
+
+More to this, the type of sequencer can be changed without stoping the sequence
+by using **ad7490_change_seq** API, or stopped by using the **ad7490_stop_seq**
+API.
+
+After starting the sequencer, in order to read the channel values, one should
+use te **ad7490_read_seq** API which handles all the required operations
+related to the sequencer and returning all requested channel values
+in their ascending index order.
+
+AD7490 Driver Initialization Example
+------------------------------------
+
+.. code-block:: bash
+
+	struct ad7490_desc *ad7490;
+	struct ad7490_init_param ad7490_ip = {
+		.spi_param =  {
+			.device_id = 4,
+			.max_speed_hz = 100000,
+			.mode = NO_OS_SPI_MODE_2,
+			.chip_select = 0,
+			.platform_ops = &max_spi_ops,
+			.extra = &max_spi_extra,
+		},
+		.op_mode = AD7490_MODE_NORMAL,
+		.vdd = AD7490_VDD_5V,
+	};
+	int ret, i;
+	int16_t val;
+
+	ret = ad7490_init(&ad7490, &ad7490_ip);
+	if (ret)
+		goto error;
+
+AD7490 no-OS IIO Support
+------------------------
+
+The AD7490 IIO driver comes on top of AD7490 driver and offers support for
+interfacing IIO clients through IIO lib.
+
+IIO AD7490 Device Configuration
+-------------------------------
+
+Device Attributes
+-----------------
+
+AD7490 IIO device has a total of 2 device attributes:
+
+* range - The range can be selected as REFIN (2.5V) or 2x REFIN (5V)
+* polarity - The polarity can be selected as UNIPOLAR or BIPOLAR.
+* sampling_frequency - The sampling frequency for AD7490 (1MSPS).
+
+Device Channels
+---------------
+
+AD7490 IIO device has 16 input channels : 16 voltage channels.
+
+The voltage channels are :
+
+* Channel 0: Voltage
+* Channel 1: Voltage
+* Channel 2: Voltage
+* Channel 3: Voltage
+* Channel 4: Voltage
+* Channel 5: Voltage
+* Channel 6: Voltage
+* Channel 7: Voltage
+* Channel 8: Voltage
+* Channel 9: Voltage
+* Channel 10: Voltage
+* Channel 11: Voltage
+* Channel 12: Voltage
+* Channel 13: Voltage
+* Channel 14: Voltage
+* Channel 15: Voltage
+
+Each voltage channel has 2 attributes:
+
+* raw - the raw voltage value read from the device.
+* scale - is the scale that has to be applied to the raw value in order to obtain the converted real value in Volts. It depends on the selected range and has a specific formula.
+
+In order to obtain the value in the V units, raw has to be scaled by scale:
+
+.. code-block:: bash
+
+	converted_voltage [V] = raw * scale
+	converted_voltage [V] = raw * (range/4096)
+
+Device buffers
+--------------
+
+The AD7490 IIO devices driver supports the usage of a data buffer for reading purposes.
+
+AD7490 IIO Driver Initialization Example
+----------------------------------------
+
+.. code-block:: bash
+
+	#define DATA_BUFFER_SIZE 400
+	uint8_t iio_data_buffer[DATA_BUFFER_SIZE * 16 * sizeof(int)];
```

Comment:
sizeof(int) will typically do what you want but you might want to be more specific like sizeof(int16_t)